browser/static/js/main_zstack.js

```diff
@@ -993,11 +905,13 @@ function load_file(file) {
       channelMax = payload.channel_max;
       rawDimensions = payload.dimensions;
 
-      sx = 0;
-      sy = 0;
-      swidth = rawWidth = rawDimensions[0];
-      sheight = rawHeight = rawDimensions[1];
+      rawWidth = rawDimensions[0];
+      rawHeight = rawDimensions[1];
 
+      viewer = new CanvasView(width=rawWidth, height=rawHeight);
+
+      // TODO: does this need to be in load_file?
+      // we could probably call it right after load_file to decouple them better
```

Comment:
It doesn't need to be in load_file, but because it is reliant on the return value of the HTTP request (as are ALL of the objects we have).

One way we could try to decouple them is by passing a callback function through `load_file(file, cb)` and defining that callback function inside `startCaliban`. that way we handle all the object creation where we get the raw dimensions, but still have it defined inside our entrypoint.